contracts/strategies/PlatypusVoterProxy.sol

```diff
@@ -0,0 +1,255 @@
+        uint256 stakerFeeBips;
+        uint256 boosterFeeBips;
+        address stakerFeeReceiver;
+        address boosterFeeReceiver;
+    }
+
+    uint256 internal constant BIPS_DIVISOR = 10000;
+
+    uint256 public boosterFee;
+    uint256 public stakerFee;
+    address public stakerFeeReceiver;
+    address public boosterFeeReceiver;
+    address public constant PTP = address(0x22d4002028f537599bE9f666d1c4Fa138522f9c8);
+    IPlatypusVoter public immutable platypusVoter;
+    address public immutable devAddr;
+
+    modifier onlyDev() {
+        require(msg.sender == devAddr, "PlatypusVoterProxy::onlyDev");
+        _;
+    }
+
+    modifier onlyStrategy(address _asset) {
+        require(strategies[_asset] == msg.sender, "PlatypusVoterProxy:onlyStrategy");
+        _;
+    }
+
+    // asset => strategies
+    mapping(address => address) public strategies;
+
+    constructor(
+        address _platypusVoter,
+        address _devAddr,
+        FeeSettings memory _feeSettings
+    ) {
+        devAddr = _devAddr;
+        boosterFee = _feeSettings.boosterFeeBips;
+        stakerFee = _feeSettings.stakerFeeBips;
+        stakerFeeReceiver = _feeSettings.stakerFeeReceiver;
+        boosterFeeReceiver = _feeSettings.boosterFeeReceiver;
+        platypusVoter = IPlatypusVoter(_platypusVoter);
+    }
+
+    function approveStrategy(address _asset, address _strategy) external onlyDev {
+        strategies[_asset] = _strategy;
+    }
+
+    function setBoosterFee(uint256 _boosterFeeBips) external onlyDev {
+        boosterFee = _boosterFeeBips;
+    }
+
+    function setStakerFee(uint256 _stakerFeeBips) external onlyDev {
+        stakerFee = _stakerFeeBips;
+    }
+
+    function setBoosterFeeReceiver(address _boosterFeeReceiver) external onlyDev {
+        boosterFeeReceiver = _boosterFeeReceiver;
+    }
+
+    function setStakerFeeReceiver(address _stakerFeeReceiver) external onlyDev {
+        stakerFeeReceiver = _stakerFeeReceiver;
+    }
+
+    function deposit(
+        uint256 _pid,
+        address _stakingContract,
+        address _pool,
+        address _token,
+        address _asset,
+        uint256 _amount
+    ) external onlyStrategy(_asset) {
+        IERC20(_token).safeApprove(_pool, _amount);
+        uint256 liquidity = IPlatypusPool(_pool).deposit(
+            address(_token),
+            _amount,
+            address(platypusVoter),
+            type(uint256).max
+        );
+        platypusVoter.safeExecute(
+            _asset,
+            0,
+            abi.encodeWithSignature("approve(address,uint256)", _stakingContract, liquidity)
+        );
+        platypusVoter.safeExecute(
+            _stakingContract,
+            0,
+            abi.encodeWithSignature("deposit(uint256,uint256)", _pid, liquidity)
+        );
+    }
+
+    function reinvestFeeBips() external returns (uint256) {
+        uint256 boostFee = 0;
+        if (boosterFee > 0 && boosterFeeReceiver > address(0) && platypusVoter.depositsEnabled()) {
```

Comment:
`platypusVoter.depositsEnabled()` nice touch

Reply:
the fee could be >0 even if deposits are not enabled. PS - not seeing this fn in platypusVoter